src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs

```diff
@@ -364,62 +362,58 @@ protected override void OnElementChanged(ElementChangedEventArgs<MediaElement> e
 		{
 			base.OnElementChanged(e);
 
+			if (e.NewElement != null)
+			{
+				SetNativeControl(avPlayerViewController.View);
+
+				Element.PropertyChanged += OnElementPropertyChanged;
+				Element.SeekRequested += MediaElementSeekRequested;
+				Element.StateRequested += MediaElementStateRequested;
+				Element.PositionRequested += MediaElementPositionRequested;
+
+				avPlayerViewController.ShowsPlaybackControls = Element.ShowsPlaybackControls;
+				avPlayerViewController.VideoGravity = AspectToGravity(Element.Aspect);
+				if (Element.KeepScreenOn)
+					SetKeepScreenOn(true);
+
+				playedToEndObserver = NSNotificationCenter.DefaultCenter.AddObserver(AVPlayerItem.DidPlayToEndTimeNotification, PlayedToEnd);
+
+				UpdateBackgroundColor();
+				UpdateSource();
+			}
+
 			if (e.OldElement != null)
```

Comment:
please move Old Element logic to be above New Element one.
We should do it for making sure we are not disposing of necessary observers

Reply:
Btw, I propose to add a new method **Dispose**

```csharp
void Dispose (ref IDisposable disposable)
{
	disposable?.Dispose();
	disposable = null;
}
```

So we can use it everywhere as
```csharp
Dispose (ref playedToEndObserver);
Dispose (ref rateObserver);
```
etc.

Reply:
Implemented